### PR TITLE
Partial excludes from hydration

### PR DESCRIPTION
**Pull request by markusheiden**
If having big entity graphs, one sometimes may want to exclude some attributes from hydration to reduce the memory footprint.

This PR bases on my prior MR.

### CODE REVIEW

File: src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     * {@link jakarta.persistence.Embeddable}s are not supported.
+     *
+     * @param entityClass
+     *            Entity class.
+     * @param attribute
+     *            Attribute name.
+     * @throws IllegalArgumentException if then entity class is not a JPA entity or the attribute does not exist.
+     * @return new instance with the exclusion.
+     */
+    public Hydrator withExclude(Class<?> entityClass, String attribute);
```
Comment (arey) [inline src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java:49]: It looks like a fluent API, nice

Comment (arey) [reply src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java:49]: Now that you've added a usage section to the `readme` (thanks a lot), I suggest you complete it by using the `withExclude` method

File: src/test/java/com/javaetmoi/core/persistence/hibernate/HydratorImplTest.java [hunk 2]
```diff
@@ -152,10 +156,26 @@ void deepHydrate_attachedEntity() {
...
+        // At this step, transaction and session are closed.
+        assertThat(hydratedEntityProxy)
+                .isEqualTo(james);
+    }
+
+    /**
+     * Tests the method {@link HydratorImpl#deepHydrate(Object)}.
+     */
+    @Test
+    void deepHydrate_withExclude() {
```
Comment (arey) [inline src/test/java/com/javaetmoi/core/persistence/hibernate/HydratorImplTest.java:170]: A test case could be added with 2 exclusions

**Comment by markusheiden** [comment]
@arey Do you mind a release? ;-)

**Comment by arey** [comment]
@markusheiden I've triggered a release
The GitHub build continues to fail : https://github.com/arey/hibernate-hydrate/actions/runs/6322310079/job/17167689945#step:6:2697
We are not alone: https://issues.sonatype.org/browse/OSSRH-51097
We are using the latest version of the `nexus-staging-maven-plugin`. I don't see how to fix it.

**Comment by arey** [comment]
The last release has failed.
I've triggered a new one https://github.com/arey/hibernate-hydrate/actions/runs/6348138901
Keeping my fingers crossed

**Comment by arey** [comment]
A new error
```
Error: [ERROR] Failed to execute goal org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13:deploy (injected-nexus-deploy) on project javaetmoi-hibernate6-hydrate: Execution injected-nexus-deploy of goal org.sonatype.plugins:nexus-staging-maven-plugin:1.6.13:deploy failed: Nexus connection problem to URL [https://oss.sonatype.org/ ]: 502 - Bad Gateway -> [Help 1]
```
Looks like https://issues.sonatype.org/browse/OSSRH-56778

**Comment by arey** [comment]
3rd trial with the `6.3.4` version (too bad for the numbering holes). Seems to be ok: https://github.com/arey/hibernate-hydrate/actions/runs/6356248978/job/17265566507

**Pull request merged**